src/shared/analytics/analytics.background.ts

```diff
@@ -348,7 +361,13 @@ export function initialize({ account }: { account: Account }) {
     willSendRequest: createAddProviderHook({ getWalletProvider }),
   });
   const handleUserId = () => mixpanelIdentify(account);
-  account.on('authenticated', () => handleUserId());
+
+  account.on('authenticated', () =>
+    // Google Analytics does not have a clear definition of a user session.
+    // Hence, we need to define what a user session means within the extension.
+    // For simplicity, we initiate a new session each time the user authenticates.
+    Promise.allSettled([handleUserId(), resetGoogleAnalyticsSessionId()])
```

Comment:
I don't get it, what does `.allSettled` achieve here? It's not being awaited nor caught

Reply:
Thats odd, doesn't make sense to me either... Let me update this part